lib/srv/app/session.go

```diff
@@ -82,6 +82,7 @@ func (s *Server) newSession(ctx context.Context, identity *tlsca.Identity, app t
 			traits:             identity.Traits,
 			log:                s.log,
 			user:               identity.Username,
+			name:               app.GetName(),
```

Comment:
Instead of passing `name`, `uri` and `publicAddr` and other app fields individually, how about just passing entire `app` and let transport get stuff it needs from it?